src/mscorlib/shared/System/Buffers/MemoryManager.cs

```diff
@@ -13,14 +13,9 @@ namespace System.Buffers
     public abstract class MemoryManager<T> : IMemoryOwner<T>, IPinnable
     {
         /// <summary>
-        /// The number of items in the Memory<typeparamref name="T"/>.
+        /// Returns a <see cref="System.Memory{T}"/>.
         /// </summary>
-        public virtual int Length => GetSpan().Length;
-
-        /// <summary>
-        /// Returns a Memory<typeparamref name="T"/>.
-        /// </summary>
-        public virtual Memory<T> Memory => new Memory<T>(this, 0, Length);
+        public virtual Memory<T> Memory => CreateMemory(GetSpan().Length);
```

Comment:
Why not just do this `new Memory<T>(this, GetSpan().Length);`?

Reply:
As example on how to use the api?

Reply:
As example on how to use the api? That constructor is only available in coreclr/corefx

Reply:
Since it is a method call, won't it require the JIT to inline it to optimize away the call?